operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -630,10 +631,31 @@ public boolean wasInspectionRun() {
 
     private boolean isShouldContinue() {
       DomainPresenceInfo cachedInfo = getExistingDomainPresenceInfo(getNamespace(), getDomainUid());
+      int currentRetryCount = Optional.ofNullable(liveInfo)
+          .map(DomainPresenceInfo::getDomain)
+          .map(Domain::getStatus)
+          .map(DomainStatus::getRetryCount)
+          .orElse(0);
+
+      String existingError = Optional.ofNullable(liveInfo)
+          .map(DomainPresenceInfo::getDomain)
+          .map(Domain::getStatus)
+          .map(DomainStatus::getMessage)
+          .orElse(null);
+
       if (cachedInfo == null || cachedInfo.getDomain() == null) {
         return true;
+      } else if (currentRetryCount > DomainPresence.getDomainPresenceFailureRetryMaxCount()) {
+        LOGGER.fine("Stop introspection retry - exceeded configured domainPresenceFailureRetryMaxCount: "
+            + DomainPresence.getDomainPresenceFailureRetryMaxCount());
+        return false;
       } else if (isCachedInfoNewer(liveInfo, cachedInfo)) {
         return false;  // we have already cached this
+      } else if (existingError != null && existingError.startsWith("MII Fatal Error")
```

Comment:
Is there a way we can return a fatal error more clearly? It seems brittle to rely on a specific string like this.

Reply:
I don't see how to eliminate dependence on string parsing, but to make it less specific to MII and maybe a bit less brittle, perhaps use 'contains' instead of 'starts' and look for a string like 'FatalIntrospectorError'?  What do you think @rjeberhard ?

Reply:
That would be okay, but my expectation is that the introspector is returning structured data, of which the error message is one value. The least brittle would be if there was a value for "isFatalError" or "isPermanentFailure". However, I'm okay with something like what you propose.

Reply:
I changed it to contains "FatalIntrospectorError". Eventually If we can change the introspect job and operator java code has a more flexible way to communicate other than just a return code, then we can experiment the idea of always return 0 but have some text that gives more structured data.